### PR TITLE
feat(search): autocomplete custom configuration

### PR DESCRIPTION
**Pull request by david-leifker**
* Autocomplete ranking to follow search configuration
* Add support for Autocomplete customization

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: smoke-test/cypress-dev.sh [hunk 1]
```diff
@@ -19,5 +19,7 @@ yarn install
 
 source "$DIR/set-cypress-creds.sh"
 
-npx cypress open \
-   --env "ADMIN_DISPLAYNAME=$CYPRESS_ADMIN_DISPLAYNAME,ADMIN_USERNAME=$CYPRESS_ADMIN_USERNAME,ADMIN_PASSWORD=$CYPRESS_ADMIN_PASSWORD"
+if [ "${RUN_UI:-true}" == "true" ]; then
+  npx cypress open \
+     --env "ADMIN_DISPLAYNAME=$CYPRESS_ADMIN_DISPLAYNAME,ADMIN_USERNAME=$CYPRESS_ADMIN_USERNAME,ADMIN_PASSWORD=$CYPRESS_ADMIN_PASSWORD"
+fi
```
Comment (gabe-lyons) [inline smoke-test/cypress-dev.sh:25]: intentional?

File: smoke-test/build.gradle [hunk 2]
```diff
@@ -135,6 +135,21 @@ task cypressDev(type: Exec, dependsOn: [installDev, ':metadata-ingestion:install
...
+}
+
+/**
+ * The following will install Cypress data in an already running stack.
+ */
+task cypressData(type: Exec, dependsOn: [installDev, ':metadata-ingestion:installDev']) {
+    environment 'RUN_QUICKSTART', 'false'
+    environment 'DATAHUB_KAFKA_SCHEMA_REGISTRY_URL', 'http://localhost:8080/schema-registry/api/'
+    environment 'KAFKA_BROKER_CONTAINER', 'datahub-kafka-broker-1'
+    environment 'RUN_UI', 'false'
```
Comment (gabe-lyons) [inline smoke-test/build.gradle:151]: should this change be separated out?

File: docker/profiles/docker-compose.yml [hunk 3]
```diff
@@ -1,5 +1,4 @@
 ---
-version: '3.9'
```
[the remark this answers is not in the document]
Comment (david-leifker) [reply docker/profiles/docker-compose.yml:2]: yes, its deprecated

File: smoke-test/build.gradle (hunk 2 above)
Comment (david-leifker) [reply smoke-test/build.gradle:151]: just a helper method to get more data for test easier

File: smoke-test/cypress-dev.sh (hunk 1 above)
Comment (david-leifker) [reply smoke-test/cypress-dev.sh:25]: yes, allows loading cypress data without running cypress UI. simply allows populating data without running cypress itself.

File: smoke-test/build.gradle (hunk 2 above)
Comment (david-leifker) [reply smoke-test/build.gradle:151]: Just a helper to get the cypress data loaded without running cypress

File: docker/profiles/docker-compose.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (david-leifker) [reply docker/profiles/docker-compose.yml:2]: was deprecated

**Review by gabe-lyons — APPROVED** [review]

**Pull request merged**